solutions/cf-1557B.mdx

```diff
@@ -0,0 +1,344 @@
+			<td>$\infty$</td>
+			<td>1</td>
+			<td>2</td>
+			<td>3</td>
+			<td>4</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+With this map in mind, we generate the subsets as shown:
+
+<center>
+<table>
+	<tbody>
+		<tr>
+			<th>Index:</th>
+			<th className="bg-yellow-100 dark:bg-yellow-800">0</th>
+			<th>1</th>
+			<th>2</th>
+			<th>3</th>
+			<th>4</th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>6</td>
+			<td>3</td>
+			<td>4</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+<center>
+<table>
+	<tbody>
+		<tr>
+			<th>Index:</th>
+			<th className="bg-yellow-100 dark:bg-yellow-800">0</th>
+			<th className="bg-green-100 dark:bg-green-800">1</th>
+			<th>2</th>
+			<th>3</th>
+			<th>4</th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>6</td>
+			<td>3</td>
+			<td>4</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+<center>
+<table>
+	<tbody>
+		<tr>
+			<th>Index:</th>
+			<th className="bg-yellow-100 dark:bg-yellow-800">0</th>
+			<th className="bg-green-100 dark:bg-green-800">1</th>
+			<th className="bg-green-100 dark:bg-green-800">2</th>
+			<th>3</th>
+			<th>4</th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>6</td>
+			<td>3</td>
+			<td>4</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+<center>
+<table>
+	<tbody>
+		<tr>
+			<th>Index:</th>
+			<th className="bg-yellow-100 dark:bg-yellow-800">0</th>
+			<th className="bg-green-100 dark:bg-green-800">1</th>
+			<th className="bg-green-100 dark:bg-green-800">2</th>
+			<th className="bg-red-100 dark:bg-red-800">3</th>
+			<th>4</th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>6</td>
+			<td>3</td>
+			<td>4</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+<center>
+<table>
+	<tbody>
+		<tr>
+			<th>Index:</th>
+			<th className="bg-yellow-100 dark:bg-yellow-800">0</th>
+			<th className="bg-green-100 dark:bg-green-800">1</th>
+			<th className="bg-green-100 dark:bg-green-800">2</th>
+			<th className="bg-red-100 dark:bg-red-800">3</th>
+			<th className="bg-blue-100 dark:bg-blue-800">4</th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>6</td>
+			<td>3</td>
+			<td>4</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+Thus, we need 4 subarrays at minimum.
+
+**Time Complexity:** $\mathcal{O}(N\log N)$
+
+<Spoiler title="Time Complexity Proof">
+Sorting Requires $\mathcal{O}(N\log N)$, and repeated map insertions also require $\mathcal{O}(N\log N)$. Hence the overall time complexity is $\mathcal{O}(N\log N)$.
+</Spoiler>
+	
+<LanguageSection>
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+using namespace::std;
+
+typedef long long ll;
+#define pb push_back
+#define fastio ios::sync_with_stdio(0); cin.tie(0)
+#define MOD 1000000007
+#define FF first
+#define SS second
```

Comment:
can you pplease not use these macros

Reply:
These are allowed though... I thought they were according to the guide for macros

Reply:
the fastio one (along with the mod) isn't

Reply:
Fixed